### PR TITLE
Review and confirmation page for RJSF 1990

### PR DESCRIPTION
**Pull request by jbalboni**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2821

Makes some review page and confirmation page updates to the 1990.

Format dates as month/year:
screen shot 2017-08-24 at 2 39 40 pm

Fixed bug that was hiding fields that use expandUnderCondition:
screen shot 2017-08-24 at 2 35 38 pm

Confirmation page:
screen shot 2017-08-24 at 1 50 24 pm

### CODE REVIEW

File: src/js/edu-benefits/1990-rjsf/BenefitsRelinquishmentField.jsx [hunk 1]
```diff
@@ -84,8 +84,33 @@ export default class BenefitsRelinquishmentField extends React.Component {
...
-      registry
+      onChange,
+      registry,
+      required,
+      name,
+      formContext
     } = this.props;
+
+    const benefitsRelinquished = _.get('benefitsRelinquished', formData);
+    if (formContext.reviewMode &&
```
Comment (jbalboni) [inline src/js/edu-benefits/1990-rjsf/BenefitsRelinquishmentField.jsx]: I'm not a huge fan of adding these conditionals, but it's been the easiest way so far for custom fields (vs custom widgets, where we can use `reviewWidget`)

File: test/util/nightwatch-commands/fillMonthYear.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/**
```
Comment (jbalboni) [inline test/util/nightwatch-commands/fillMonthYear.js]: I made this and then realized I didn't need it right now, but we will later.

File: src/js/common/schemaform/review/ObjectField.jsx [hunk 3]
```diff
@@ -102,13 +102,14 @@ class ObjectField extends React.Component {
 
     const renderedProperties = this.orderAndFilterProperties(properties)
       .map((objectFields, index) => {
-        const firstField = objectFields[0];
         // show all the fields only if the first one is truthy, since more than one field
         // means this is an expanding group
-        if (objectFields.length > 1 && !!formData[firstField]) {
+        const [first, ...rest] = objectFields;
```
Comment (jbalboni) [inline src/js/common/schemaform/review/ObjectField.jsx]: This logic copied from the normal ObjectField, which is what should have happened when I originally wrote it.

File: src/js/common/schemaform/review/ObjectField.jsx [hunk 4]
```diff
@@ -102,13 +102,14 @@ class ObjectField extends React.Component {
 
     const renderedProperties = this.orderAndFilterProperties(properties)
       .map((objectFields, index) => {
-        const firstField = objectFields[0];
         // show all the fields only if the first one is truthy, since more than one field
         // means this is an expanding group
```
Comment (cvalarida) [inline src/js/common/schemaform/review/ObjectField.jsx]: This comment doesn't seem to be pertinent anymore...is it?

File: src/js/edu-benefits/1990-rjsf/containers/ConfirmationPage.jsx [hunk 5]
```diff
@@ -30,21 +33,49 @@ class ConfirmationPage extends React.Component {
...
 
     const docExplanation = this.state.isExpanded
       ? (<div className="usa-accordion-content">
-        <p>In the future, you might need a copy of your DD 2863 (National Call to Service (NCS) Election of Options).</p>
+        <p>In the future, you might need:</p>
+        <ul>
+          <li>Your reserve kicker</li>
+          <li>Documentation of additional contributions that would increase your monthly benefits.</li>
+        </ul>
         <p>Documents can be uploaded using the <a href="https://gibill.custhelp.com/app/utils/login_form/redirect/account%252">GI Bill site</a>.</p>
```
Comment (cvalarida) [inline src/js/edu-benefits/1990-rjsf/containers/ConfirmationPage.jsx]: Is the `%252` necessary for some reason? I know it's not a new change, but...it's odd.

File: test/common/schemaform/review/ObjectField.unit.spec.jsx [hunk 6]
```diff
@@ -123,7 +123,7 @@ describe('Schemaform review: ObjectField', () => {
     expect(tree.everySubTree('.form-review-panel-page-header-row')).not.to.be.empty;
     expect(tree.subTree('.form-review-panel-page-header').text()).to.be.empty;
   });
-  it('should hide expand under items when false', () => {
+  it('should hide expand under items when collapsed', () => {
```
Comment (cvalarida) [inline test/common/schemaform/review/ObjectField.unit.spec.jsx]: This seems like an odd change...Why?

File: test/util/nightwatch-commands/fillMonthYear.js (hunk 2 above)
Comment (cvalarida) [reply test/util/nightwatch-commands/fillMonthYear.js]: We could just be able to adapt `.fillDate()` to take a partial, too.

**Review by cvalarida — APPROVED** [review]

**Review by cehsu — APPROVED** [review]
This looks good to me.

File: test/common/schemaform/review/ObjectField.unit.spec.jsx (hunk 6 above)
Comment (jbalboni) [reply test/common/schemaform/review/ObjectField.unit.spec.jsx]: We used to hide expand under items when the value of the property being expanded under was false. Now, we test against the value of `expandUnderCondition` as well, with that logic living in the reducer code. The actual logic in this component is just to check against `ui:collapsed`.

**Pull request merged**